rosidl_typesupport_opensplice_cpp/resource/srv__type_support.cpp.em

```diff
@@ -146,14 +149,12 @@ public:
                "the data_values and info_seq were not obtained from this "
                "@(__dds_sample_type_prefix)@(suffix)_DataReader";
       case DDS::RETCODE_OK:
-        *taken = true;
-        return nullptr;
+        break;
```

Comment:
Oh you mean that nothing else happens after the switch, so we might as well return nullptr here. I did it because there is another spot with similar code that does do extra work after the switch statement but before returning. There's no particular reason. Maybe so someone doesn't add stuff after the switch but before the return and not notice the function returns here.